deps/rabbit/src/rabbit_amqqueue.erl

```diff
@@ -2061,3 +2110,53 @@ is_queue_args_combination_permitted(Durable, Exclusive) ->
         true ->
             rabbit_deprecated_features:is_permitted(transient_nonexcl_queues)
     end.
+
+-spec kill_queue_hard(node(), name()) -> ok.
+kill_queue_hard(Node, QRes = #resource{kind = queue}) ->
+    kill_queue_hard(Node, QRes, boom).
+
+-spec kill_queue_hard(node(), name(), atom()) -> ok.
+kill_queue_hard(Node, QRes = #resource{kind = queue}, Reason) ->
+    case kill_queue(Node, QRes, Reason) of
+        crashed -> ok;
+        stopped -> ok;
+        _NewPid -> timer:sleep(?KILL_QUEUE_DELAY_INTERVAL),
+                   kill_queue_hard(Node, QRes, Reason)
+    end.
+
+-spec kill_queue(node(), name()) -> pid() | crashed.
+kill_queue(Node, QRes = #resource{kind = queue}) ->
+    kill_queue(Node, QRes, boom).
+
+-spec kill_queue(node(), name(), atom()) -> pid() | crashed | stopped.
+kill_queue(Node, QRes = #resource{kind = queue}, Reason = shutdown) ->
+    Pid1 = pid_or_crashed(Node, QRes),
+    exit(Pid1, Reason),
```

Comment:
This exits but then tries to await a pid and return a value. What am I missing?

Reply:
it exits the target pid (not calling pid) then just waits for expected state to be updated in mnesia